src/build_queue.rs

```diff
@@ -48,6 +47,31 @@ impl BuildQueue {
         }
     }
 
+    pub fn last_seen_reference(&self) -> Result<Option<Oid>> {
+        let mut conn = self.db.get()?;
+        if let Some(value) = get_config(&mut conn, ConfigName::LastSeenIndexReference)?.as_str() {
+            match Oid::from_str(value) {
+                Ok(oid) => return Ok(Some(oid)),
+                Err(err) => {
+                    log::error!("queue locked because of invalid last_seen_index_reference \"{}\" in database: {}", value, err);
+                    self.lock()?;
+                    return Ok(None);
```

Comment:
Shouldn't this return an error instead of discarding it?

Reply:
I was kind of coming from the `watch_registry` method where I want to keep the registry watcher running, but locked, after this happened. 

So we would then manually remove / fix the ref and unlock the queue. 

But I can see that this could be confusing behavior for `last_seen_reference` and moved the logic into `watch_registry`